buildSrc/src/main/java/org/apache/lucene/gradle/WrapperDownloader.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.lucene.gradle;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.math.BigInteger;
+import java.net.URL;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+
+public class WrapperDownloader {
+    public static void main(String[] args) throws Exception {
+        if (args.length != 3) {
+            System.err.println("Usage: java WrapperDownloader.java <version> <destination> <checksum>");
+            System.exit(1);
+        }
+
+        File jar = new File(args[1]);
+        if (!jar.exists()) {
+            final URL url = new URL("https://github.com/gradle/gradle/raw/v" + args[0] + "/gradle/wrapper/gradle-wrapper.jar");
+
+            // As of v6.0.1 the wrapper is approximately 60K
+            // Can increase this is gradle wrapper ever goes beyond 100K, but keep a safety check
+            final int maxSize = 100 * 1024;
```

Comment:
Do you think we really need this check? I'd say simplify and just url.openStream().transferTo(outputStream).

Reply:
I'd like to have the check, yes. Got burned by something like this once, and never again :)

Reply:
I'm afraid to ask what happened. :) I'd bump the limit to a reasonable 500K just in case the wrapper grows and we forget about the check...